ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -688,6 +688,16 @@ public class KsqlConfig extends AbstractConfig {
       + "lifespan (if present) and the value of this config. If this config is set to 0, then "
       + "ksqlDB will not close websockets even if the token has an expiration time.";
 
+
+  public static final String KSQL_SELF_JOIN_OPTIMIZATION_ENABLE
```

Comment:
I added this config as a safe-guard so that we can turn the rewriting off in case things go wrong in the future